moveit_ros/hybrid_planning/local_planner/trajectory_operator_plugins/src/simple_sampler.cpp

```diff
@@ -77,14 +80,18 @@ bool SimpleSampler::reset()
   reference_trajectory_->clear();
   return true;
 }
-robot_trajectory::RobotTrajectory SimpleSampler::getLocalTrajectory(const moveit::core::RobotState& current_state)
+moveit_msgs::action::LocalPlanner::Feedback
+SimpleSampler::getLocalTrajectory(const moveit::core::RobotState& current_state,
+                                  robot_trajectory::RobotTrajectory& local_trajectory)
 {
-  robot_trajectory::RobotTrajectory local_trajectory(reference_trajectory_->getRobotModel(),
-                                                     reference_trajectory_->getGroupName());
+  // Delete previous local trajectory
+  local_trajectory.clear();
+
+  // Determine current local trajectory based on configured behavior
   if (pass_through_)
   {
     // Use reference_trajectory as local trajectory
-    local_trajectory.swap(*reference_trajectory_);
+    local_trajectory = robot_trajectory::RobotTrajectory(*reference_trajectory_);
```

Comment:
So, this constructs a new robot trajectory each time, so clearing `local_trajectory` would be redundant. But I'd guess that clearing and then copying the waypoints over would be more efficient.